src/test/java/programminglife/parser/GraphParserTest.java

```diff
@@ -122,4 +137,112 @@ public void update(Observable o, Object arg) {
             }
         }
     }
+
+    @Test
+    public void parseSegmentNotStartingS() {
+        try {
+            graphParser.parseSegment("Fiets");
+            fail("This should not be parsed");
+        } catch (Exception e) {
+            assertEquals(UnknownTypeException.class, e.getClass());
+            assertEquals("Line (Fiets) is not a segment", e.getMessage());
+        }
+    }
```

Comment:
These should use `@Test(expected = UnnownTypeException.class)` as that is more robust and removes anyneeds to use `fail()`